src/containers/Projects/index.js

```diff
@@ -125,17 +166,19 @@ export class Projects extends Component {
 }
 
 Projects.propTypes = {
+  pathname: PropTypes.string.isRequired,
   isFetching: PropTypes.bool.isRequired,
   errors: PropTypes.array,
-  projects: PropTypes.array.isRequired,
+  tree: PropTypes.object.isRequired,
   dispatch: PropTypes.func.isRequired,
   theme: PropTypes.object.isRequired,
 }
 
 export default connect(
   state => ({
+    pathname: state.routing.locationBeforeTransitions.pathname,
     isFetching: state.projects.isFetching,
     errors: state.projects.errors || [],
-    projects: state.projects.tree || [],
+    tree: state.projects.tree || { groups: [], projects: [] },
```

Comment:
Let us get rid of the default arguments. And split tree into two props instead. Gives you more type safety.